File: lc_soln/0217_contains_duplicate/solution.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> seen;
        for (const int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.insert(num);
        }
        return false;
    }

};

int main() {
    return 0;
}

File: lc_soln/0217_contains_duplicate/solution.rs
use std::collections::HashSet;

struct Solution;

impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::new();
        for num in nums {
            if seen.contains(&num) {
                return true;
            }
            seen.insert(num);
        }
        false
    }
}

fn main() {
    println!("{}", Solution::has_duplicate(vec![1, 2, 3, 3]));
}
